lib/relaton_nist/hit_collection.rb

```diff
@@ -241,18 +193,17 @@ def docidentifier(json) # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticCom
     # @return [Array<Hash>] selected data
     #
     def select_data # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength,Metrics/PerceivedComplexity
-      ref = "#{refparts[:series]} #{refparts[:code]}"
       d = Date.strptime year, "%Y" if year
-      statuses = %w[draft-public draft-prelim]
+      ref = "#{@pubid_ref.serie.serie.sub(/^NIST\s|\sPUB$/, '')} #{@pubid_ref.code}" # TODO: Pubid shouldn't have PUB
       PubsExport.instance.data.select do |doc|
         next unless match_year?(doc, d)
 
-        if @opts[:stage]&.include?("PD")
-          next unless statuses.include? doc["status"]
+        if @pubid_ref.stage&.type == "pd"
+          next unless doc["status"] == "draft-public"
         else
```

Comment:
Can't we move the detection of `draft-public` to an earlier stage to store in pubid?

Reply:
This is a loop through all the items in pubs-export dataset. The Parslet, used in Pubid gems, is quite slow. So, parsing every ID in the dataset will slow down serachig. In this loop a few elements seelcted by code and stage. Next, IDs of the selected items are parsed and compared to reference pubid. Does it make sense?